SPARQLWrapper/SPARQLExceptions.py

```diff
@@ -28,18 +30,18 @@ class SPARQLWrapperException(Exception):
 
     msg = "an exception has occurred"
 
-    def __init__(self, response=None):
+    def __init__(self, response: Optional[bytes] = None):
         """
         :param string response: The server response
         """
-        if response:
-            formatted_msg = "%s: %s. \n\nResponse:\n%s" % (
+        if response is None:
+            formatted_msg = "%s: %s." % (self.__class__.__name__, self.msg)
+        else:
+            formatted_msg = "%s: %s. \n\nResponse:\n%r" % (
                 self.__class__.__name__,
                 self.msg,
                 response,
             )
```

Comment:
While I understand the reason for this change, but type checking passes fine without it, and it would be best to keep the scope of a PR as small as possible, and defer changes outside of the scope to other PRs.

There are a couple of changes in this PR that is similar in nature, they all seem good to me, but they are not needed for type checking to pass, and it would be best to make them separately. It makes this PR easier to review and the other changes simpler to evaluate.

Reply:
For [RDFLib/rdflib](https://github.com/RDFLib/rdflib), when I add type hints, I make PRs that have no runtime impact, even though there are sometimes things that I notice is broken and could be fixed. Adding a TODO/FIXME is also an option.

Reply:
I'm okay with this if another maintainer is okay with it, but our general rule for PRs are that runtime changes (as this) should be accompanied with tests.

Reply:
I think this rule is also somewhat flexible, but if we have type annotations, runtime changes which are very safe and runtime changes that probably should be tested in one PR then it makes it harder to justify.

Reply:
934e334